### ISSUE TITLE
Weird synchrinization blocks

### ISSUE THREAD

**Issue by kofemann** [stleary/JSON-java#765]
By looking at the code I noticed some weird synchronization blocks, like:

```java
    public String toString(int indentFactor) throws JSONException {
        StringWriter w = new StringWriter();
        synchronized (w.getBuffer()) {
            return this.write(w, indentFactor, 0).toString();
        }
    }
```

https://github.com/stleary/JSON-java/blob/master/src/main/java/org/json/JSONObject.java#L2573

Obviously, synchronizing on a newly created object makes no sense.

**Comment by johnjaylward** [comment]
Yes, all instances of that seem to be pretty useless. I don't see how any other threads would be able to write to the writers and require synchronization.

- https://github.com/stleary/JSON-java/blob/master/src/main/java/org/json/JSONObject.java#L2180
- https://github.com/stleary/JSON-java/blob/master/src/main/java/org/json/JSONObject.java#L2573
- https://github.com/stleary/JSON-java/blob/master/src/main/java/org/json/JSONArray.java#L1622

If anything, I would have expected synchronization locks in the more generic `write` methods that take a `Writer` as a parameter.

**Comment by stleary** [comment]
This has been in the code for over 10 years. But agree it seems unnecessary. No objection if someone wants to remove these instances of synchronization.

**Comment by stleary** [comment]
Fixed in #774
